src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -531,6 +538,52 @@ public void delete(String modelId, ActionListener<DeleteModelResponse> listener)
                     return;
                 }
 
+                // Get index mappings to check if model is being used by any indices
+                getMappings(modelId, getMappingsStep);
+            }, listener::onFailure);
+
+            getMappingsStep.whenComplete(getMappingsResponse -> {
+                String modelIdStringInMapping = String.join("model_id=", modelId);
+                Map<String, MappingMetadata> mappings = getMappingsResponse.getMappings();
+                for (Map.Entry<String, MappingMetadata> entry : mappings.entrySet()) {
```

Comment:
Looking at this code it desperately needs something like `Jackson-databind` as a dependency. Its not worth maintaining and parsing Json like this when there are opensource libraries written for this specific purpose. can you please look for something similar in Opensearch?

Reply:
I think guava is better than jackson in terms of vulnerabilities, I can look into using that